src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -203,6 +241,7 @@ private void replaceIndex(final ActionRequest request, final String oldIndexName
                 }
 
                 if (ar instanceof IndexRequest) {
+                    createIndexAndAliasIfAbsent(newIndexName);
```

Comment:
This looks interesting. Before this change, does the logic not even check whether the index exists before starting indexing? Looks like a bug to me if there's really no this check. Have we faced any problem because of this at all?

Reply:
ES implicitly (auto) creates index if it does not exist.